src/typings/api.d.ts

```diff
@@ -0,0 +1,137 @@
+import compareVersions from "compare-versions";
+import { App, Component } from "obsidian";
+import { FullIndex } from "../data";
+import { DataObject, Link, LiteralValue, Values, Task } from "../data/value";
+import { DataviewSettings } from "../settings";
+import { DataArray } from "../api/data-array";
+import { BoundFunctionImpl } from "../expression/functions";
+import { Context } from "../expression/context";
+import { DateTime, Duration } from "luxon";
+import * as Luxon from "luxon";
+
+/** Asynchronous API calls related to file / system IO. */
+export interface DvIOAPIInterface {
```

Comment:
What do we gain specifically from independent typings (as opposed to just exporting `DataviewApi` directly)?

Reply:
define a new interface separated from class DataviewApi can better check exposed api members and breaking API changes (it's achieved with [`implement` keyword](https://blog.logrocket.com/interfaces-in-typescript-what-are-they-and-how-do-we-use-them-befbc69b38b3/), now class `DataviewApi` implements `DvAPIInterface`, `DataviewIOApi` implement `DvIOAPIInterface` as well)

Reply:
For example, if you changed the arg definition of a function in the class before, you may forget to mark it as a breaking change. Using a interface enforce the class to implement the current version of API and gives a warning if you change that arg.

Reply:
Ah, okay, I understand - I'm fine with that so I can remember to include breaking changes in release notes.